packages/elements/src/react/common/form/index.tsx

```diff
@@ -546,7 +580,43 @@ const Input = React.forwardRef<React.ElementRef<typeof RadixControl>, FormInputP
       }),
     );
 
-    const field = useInput(props);
+    if (signInRouterRef && passkeyAutofillProp) {
+      return (
+        <SignInInput
+          ref={forwardedRef}
+          {...props}
+        />
+      );
+    }
+
+    if (passkeyAutofillProp) {
+      throw new ClerkElementsRuntimeError(`<Input passkeyAutofill> can only be used inside <SignIn>.`);
+    }
+
+    return (
+      <CommonInput
+        ref={forwardedRef}
+        {...props}
+      />
+    );
+  },
+);
+
+Input.displayName = INPUT_NAME;
+
+const SignInInput = React.forwardRef<React.ElementRef<typeof RadixControl>, FormInputProps>(
+  (props: FormInputProps, forwardedRef) => {
+    const signInRouterRef = SignInRouterCtx.useActorRef(true);
+    const passkeyAutofillSupported = useSignInPasskeyAutofill();
+
+    React.useEffect(() => {
+      if (passkeyAutofillSupported) {
+        signInRouterRef?.send({ type: 'AUTHENTICATE.PASSKEY.AUTOFILL' });
+      }
+    }, [passkeyAutofillSupported, signInRouterRef]);
```

Comment:
Can this be in the `useInput()` hook? Ideally the components here are just thin wrappers around the hook, which contains the business logic

Reply:
Probably yes, but if felt like something only this component should care about.

because we are using refs here and not the context, it will not error, but i thought it would ok to not litter useInput with sign in specific logic.

Reply:
Turns out with the current setup this is a bit hard. `useSignInPasskeyAutofill` will use the `useSelector` underneath which would error when the SignInRouter context is not available.